addons/account/models/partner.py

```diff
@@ -205,7 +205,7 @@ class ResPartner(models.Model):
     @api.multi
     def _credit_debit_get(self):
         tables, where_clause, where_params = self.env['account.move.line']._query_get()
-        where_params = [tuple(self.ids)] + where_params
+        where_params = [tuple(self.ids)] + where_params + [self.env.user.company_id.id]
```

Comment:
Since a partner could be shared through multiple companies, are you sure it's correct in a functional point of view?

Reply:
credit and debit are computed fields, non stored. So they'll adapt according to the context. No problems here.
I agree this change may be sensitive, although I'm sure there is something missing about companies, because the SQL wouldn't discriminate across them, which is plain wrong anyway.
So, there are two solutions:
- these fields compute their amount across all companies that the user can see
   => I don't think it's right, because [my opw] and because if you would have a link to the invoices that the number refers to, you would have an access error if the invoice is not in the current company.
- they compute their amount as my fix proposes. And i guess it complies with multi-company flows
   => it changes the semantics of it but has the advantage of partitioning the business between companies. A-priori, when a user is in one company, he/she wants to deal with the business of that company, and shouldn't care about other companies
The field has been added by v9 yeahhhhhh, so I guess @qdp-odoo has an opinion ;)

Reply:
I agree with @kebeclibre here